test/test_irclib.py

```diff
@@ -579,6 +579,105 @@ def testChghost(self):
             command='CHGHOST', args=['bar2', 'baz2']))
         self.assertEqual(st.nickToHostmask('foo'), 'foo!bar2@baz2')
 
+    def testNickToAccountBaseJoin(self):
+        st = irclib.IrcState()
+
+        st.addMsg(self.irc, ircmsgs.join('#foo', prefix='foo!bar@baz'))
+        st.addMsg(self.irc, ircmsgs.join('#foo', prefix='bar!baz@qux'))
+        with self.assertRaises(KeyError):
+            st.nickToAccount('foo')
+            st.nickToAccount('bar')
+
+
+    def testNickToAccountExtendedJoin(self):
+        st = irclib.IrcState()
+        st.capabilities_ack.add('extended-join')
+
+        st.addMsg(self.irc, ircmsgs.IrcMsg(
+            command='JOIN', prefix='foo!bar@baz',
+            args=['#foo', 'account1', 'real name1']))
+        st.addMsg(self.irc, ircmsgs.IrcMsg(
+            command='JOIN', prefix='bar!baz@qux',
+            args=['#foo', 'account2', 'real name2']))
+        st.addMsg(self.irc, ircmsgs.IrcMsg(
+            command='JOIN', prefix='baz!qux@quux',
+            args=['#foo', '*', 'real name3']))
+        self.assertEqual(st.nickToAccount('foo'), 'account1')
+        self.assertEqual(st.nickToAccount('bar'), 'account2')
+        self.assertIsNone(st.nickToAccount('baz'))
+        with self.assertRaises(KeyError):
+            st.nickToAccount('qux')
+
+        # QUIT erases the entry
+        with self.subTest("QUIT"):
+            st2 = st.copy()
+            st2.addMsg(self.irc, ircmsgs.quit(prefix='foo!bar@baz'))
+            with self.assertRaises(KeyError):
+                st2.nickToAccount('foo')
+            self.assertEqual(st2.nickToAccount('bar'), 'account2')
+            self.assertEqual(st.nickToAccount('foo'), 'account1')
```

Comment:
Check `st2` instead of `st`?

Reply:
nope, this is meant to check `st` isn't affected by a message added to `st2`

Reply:
I added a comment to clear the confusion